c39778366.lua

```diff
@@ -41,14 +41,20 @@ end
 function c39778366.operation(e,tp,eg,ep,ev,re,r,rp)
 	local ft=Duel.GetLocationCount(tp,LOCATION_MZONE)
 	if ft<=0 then return end
-	local g=Duel.GetChainInfo(0,CHAININFO_TARGET_CARDS)
-	local sg=g:Filter(Card.IsRelateToEffect,nil,e)
-	if sg:GetCount()>1 and Duel.IsPlayerAffectedByEffect(tp,59822133) then return end
-	if sg:GetCount()>ft then
+	local g=Duel.GetChainInfo(0,CHAININFO_TARGET_CARDS):Filter(Card.IsRelateToEffect,nil,e)
+	if g:GetCount()>1 and Duel.IsPlayerAffectedByEffect(tp,59822133) then return end
+	local sg=nil
+	if g:GetCount()>ft then
 		Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_SPSUMMON)
-		sg=sg:Select(tp,ft,ft,nil)
+		sg=g:Select(tp,ft,ft,nil)
+		g:Sub(sg)
+	else
+		sg=g
 	end
 	Duel.SpecialSummon(sg,0,tp,tp,false,false,POS_FACEUP)
+	if g:GetCount()>0 then
```

Comment:
Why does this card use different logic from the other cards? And this will make the monsters spsummoned be sent to grave immediately when g:GetCount()<=ft.